app/screens/GettingCallScreen/GettingCallScreen.tsx

```diff
@@ -1,204 +1,103 @@
-import { useRef, useState, useEffect } from 'react';
-import { View, TouchableOpacity } from 'react-native';
-import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
+import { PeerConnectionContext } from 'providers/CallVideoProvider';
+import { useState, useEffect, useContext } from 'react';
 import {
-  MediaStream,
-  RTCPeerConnection,
   EventOnAddStream,
-  RTCIceCandidate,
+  EventOnCandidate,
+  MediaStream,
   RTCSessionDescription,
+  RTCSessionDescriptionType,
 } from 'react-native-webrtc';
-import GetStreams from 'utils/getStreams';
+import { useSelector } from 'react-redux';
 
-import { palette } from '@Themes/index';
-import firestore, { FirebaseFirestoreTypes } from '@react-native-firebase/firestore';
+import { Video } from '@Components/index';
+import { SOCKET_EVENTS } from '@Constants/index';
+import { useMediaStream } from '@Hooks/useMediaStream';
+import { WebSocketContext } from '@Providers/index';
+import { getGroupIdSelector, getNewOfferSelector } from '@Stores/callVideo';
 
-import { styles } from './GettingCallScreenStyles';
 import { GettingCall } from './components/GettingCall/GettingCall';
-import { Video } from './components/Video/Video';
-
-const configuration = { iceServers: [{ url: 'stun:stun.l.google.com:19302' }] };
 
 export const GettingCallScreen = () => {
-  const [localStream, setLocalStream] = useState<MediaStream | null>();
-  const [remoteStream, setRemoteStream] = useState<MediaStream | null>();
+  const socket = useContext(WebSocketContext);
+  const [localStream, setLocalStream] = useState<MediaStream | null>(null);
+  const [remoteStream, setRemoteStream] = useState<MediaStream | null>(null);
   const [gettingCall, setGettingCall] = useState(false);
-  const pc = useRef<RTCPeerConnection>();
 
-  const connecting = useRef(false);
+  const newOffer = useSelector(getNewOfferSelector);
+  const groupId = useSelector(getGroupIdSelector);
+  const peerConnection = useContext(PeerConnectionContext);
 
-  const setupWebrtc = async () => {
-    pc.current = new RTCPeerConnection(configuration);
+  const getMediaStream = useMediaStream();
 
-    const stream = await GetStreams.getStream();
-
-    if (stream) {
-      setLocalStream(stream);
-      pc.current.addStream(stream);
+  const handleRemoteDescription = async (newDescription: RTCSessionDescriptionType) => {
```

Comment:
Hung, this function is use multiple times, in usePeerConnection. Please create another hook for it